pkg/http-server/start.go

```diff
@@ -29,7 +32,15 @@ import (
 
 // Start initializes api routes and starts http server
 func Start(port, configFile, certFile, privateKeyFile string) {
+	logger := logging.GetDefaultLogger() // new logger
 
+	if privateKeyFile != "" || certFile != "" {
+		logger.Debugf("certfile is %s, privateKeyFile is %s", certFile, privateKeyFile)
+
+		if err := validateFiles(privateKeyFile, certFile); err != nil {
+			logger.Fatal(err)
+		}
+	}
```

Comment:
Good change!

This piece of code can be moved into it's own function, this function can have more server related validations in the future.

Also, we can associate the newly created function with `APIServer{}`, since these are API server validations.